### PR TITLE
saving 3D points objects along with board and pts id + proper 3D obje…

### PR DESCRIPTION
**Pull request by rameau-fr**
saving 3D points objects along with board and pts id + proper 3D object calib display (python)

### CODE REVIEW

File: src/Calibration.cpp [hunk 1]
```diff
@@ -307,22 +307,36 @@ void Calibration::saveCamerasParams() {
...
  * Export 3D points constituting the objects.
- *
+ * Format: X Y Z board_id pts_id
  */
 void Calibration::save3DObj() {
   std::string save_path_object = save_path_ + "calibrated_objects_data.yml";
   cv::FileStorage fs(save_path_object, cv::FileStorage::WRITE);
 
   for (const auto &it_obj : object_3d_) {
+    unsigned int a = 0;
```
Comment (BAILOOL) [inline src/Calibration.cpp:317]: a more descriptive variable name would be better

File: src/Calibration.cpp [hunk 2]
```diff
@@ -307,22 +307,36 @@ void Calibration::saveCamerasParams() {
...
+        for (int i = 0; i < board_ptr->nb_pts_; i++) {
+          std::pair<int, int> board_id_pts_id = std::make_pair(board_idx, i);
+          cv::Point3f curr_pts =
+              cur_object
+                  ->pts_3d_[cur_object->pts_board_2_obj_[board_id_pts_id]];
+          pts_mat.at<float>(0, a) = curr_pts.x;
+          pts_mat.at<float>(1, a) = curr_pts.y;
+          pts_mat.at<float>(2, a) = curr_pts.z;
+          pts_mat.at<float>(3, a) = float(board_idx);
+          pts_mat.at<float>(4, a) = float(board_id_pts_id.second);
```
Comment (BAILOOL) [inline src/Calibration.cpp:336]: let's use `static_cast<float>()` instead of the older way

File: src/Calibration.cpp [hunk 3]
```diff
@@ -307,22 +307,36 @@ void Calibration::saveCamerasParams() {
...
-    for (int i = 0; i < obj_nb_pts; i++) {
-      cv::Point3f curr_pts = cur_object->pts_3d_[i];
-      pts_mat.at<float>(0, i) = curr_pts.x;
-      pts_mat.at<float>(1, i) = curr_pts.y;
-      pts_mat.at<float>(2, i) = curr_pts.z;
+    cv::Mat pts_mat(5, obj_nb_pts, CV_32FC1);
+    for (const auto &it_board : cur_object->boards_) {
+      auto board_ptr = it_board.second.lock();
+      if (board_ptr) {
+        int board_idx = board_ptr->board_id_;
```
Comment (BAILOOL) [inline src/Calibration.cpp:325]: `const int` as we do not want to accidentally modify this variable

File: src/Calibration.cpp (hunk 1 above)
Comment (rameau-fr) [reply src/Calibration.cpp:317]: called it: "obj_pts_count"

File: src/Calibration.cpp (hunk 2 above)
Comment (rameau-fr) [reply src/Calibration.cpp:336]: Indeed!

File: src/Calibration.cpp (hunk 3 above)
Comment (rameau-fr) [reply src/Calibration.cpp:325]: do you mean `const int board_idx` ?

Comment (BAILOOL) [reply src/Calibration.cpp]: Yes

**Review by BAILOOL — APPROVED** [review]

**Pull request merged**